src/DataAccessLayer/Repositories/TicketsRepository.cs

```diff
@@ -69,21 +70,36 @@ public TicketsRepository([NotNull] IDalSettings settings)
             }
         }
 
-        public async Task<int> CreateTicket(TicketDalDtoModelRequest ticket)
+        public async Task<Tuple<StoredProcedureExecutionResult, int>> CreateTicket(TicketDalDtoModelRequest ticket)
         {
-            using (SqlConnection connection = new SqlConnection(_settings.ConnectionString))
+            try
             {
-                int id = await connection.ExecuteScalarAsync<int>(
-                    "AddTicket",
-                    new
+                using (SqlConnection connection = new SqlConnection(_settings.ConnectionString))
+                {
+                    int id = await connection.ExecuteScalarAsync<int>(
+                        "AddTicket",
+                        new
                         {
                             UserId = ticket.UserId,
                             PriceId = ticket.PriceId,
                             TicketStatus = TicketStatus.InProcess.ToString()
                         },
-                    commandType: CommandType.StoredProcedure);
+                        commandType: CommandType.StoredProcedure);
 
-                return id;
+                    return new Tuple<StoredProcedureExecutionResult, int>
+                    (
+                        StoredProcedureExecutionResult.Ok,
+                        id
+                    );
+                }
+            }
+            catch (SqlException e) when (e.Number == 2627) // Unique key violation
```

Comment:
Do you use this number in some other place? It is worth to extract it to some meaningful constant